linode/manager
# refactor: [M3-8651] - Migrate `Tabs` to `ui` package

## Description 📝
Move `Tabs` and associated components, story and tests to `ui` package

## Changes  🔄
- Move `Tabs` and associated components (`SafeTabPanel`, `TabLinkList`, `TabList`, `TabPanel`, `TabPanels`, `Tab`), story and tests to `@linode/ui` package
- Update existing imports
- Add `MemoryRouter` to ui package's `testHelpers` utilities `Options` type
- Change `Tab` type to `TabProps` to avoid naming conflicts with the `Tab` component in the barrel `index.ts` file and other parts of the manager package.

## Target release date 🗓️
N/A

## How to test 🧪
- Confirm all checks pass
- Confirm all the necessary imports are updated
- Confirm nothing on CM and Storybook breaks
   - Ensure everything works as expected wherever `Tabs` is used
   - Ensure we are not importing from `@linode/ui` in `@linode/ui` 
   - Run `rm -r packages/manager/node_modules/.cache`  + clear your browser cache before testing Storybook

## As an Author, I have considered 🤔

- 👀 Doing a self review
- ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- 🤏 Splitting feature into small PRs
- ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- 🧪 Providing/improving test coverage
- 🔐 Removing all sensitive information from the code and PR description
- 🚩 Using a feature flag to protect the release
- 👣 Providing comprehensive reproduction steps
- 📑 Providing or updating our documentation
- 🕛 Scheduling a pair reviewing session
- 📱 Providing mobile support
- ♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅
